project/app/views.py

```diff
@@ -19,6 +19,46 @@ def has_permission(self, request, view):
             print(e)
 
 
+class RequiredManager(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user.role
+        try:
+            if str(user) == 'M':
+                return Response({})
+        except Exception as e:
+            print(e)
+
+
+class RequiredEmployee(BasePermission):
```

Comment:
This Permission isn't needed since, We can handle it just by IsAuthenticated

Reply:
this is for the request where manager can add employee ... this helps to handle that he only add employee not manager or admin

Reply:
Yes, But see If admin and manager are not required then obviously the authenticated user will be an employee